src/main/java/org/spongepowered/common/mixin/core/server/management/MixinPlayerInteractionManager.java

```diff
@@ -96,77 +110,80 @@ public void onBlockRemoving(final BlockPos pos, final CallbackInfo ci) {
     }
 
     /**
-     * @author gabizou - September 5th, 2018
-     * @reason Due to the way that buckets and the like can be handled
-     * on the client, often times we need to cancel the item stack usage
-     * due to server side cancellation logic that may not exist on the client.
-     * Therefor, the cancellation of possible block changes doesn't take
-     * effect, and therefor requires telling the client to set back the item
-     * in hand.
+     * @author morpheus - December 15th, 2018
      *
-     * @param actionResult The action result returned from useItemRightClick, which if
-     *  the result is FAIL, then we should be setting the item in hand back.
-     * @param player The player
-     * @param worldIn The world
-     * @param stack The stack
-     * @param hand The hand
-     * @return The result, but we will inject a "send inventory to player packet" fi it was failed
+     * @reason Fire interact block event.
      */
-    @Redirect(
-        method = "processRightClick",
-        at = @At(
-            value = "INVOKE",
-            target = "Lnet/minecraft/util/ActionResult;getType()Lnet/minecraft/util/EnumActionResult;",
-            ordinal = 0 // We need to target only the first getType, since
-                      // there's technically two getTypes that are being caught by the slice
-
-        ),
-        slice = @Slice(
-            from = @At(
-                value = "INVOKE",
-                target = "Lnet/minecraft/item/ItemStack;getMaxItemUseDuration()I",
-                ordinal = 0 // This targets the first max use duration in the massive if statement.
-            ),
-            to = @At(
-                value = "FIELD",
-                target = "Lnet/minecraft/util/EnumActionResult;FAIL:Lnet/minecraft/util/EnumActionResult;"
-            )
-        )
-    )
-    private EnumActionResult spongeGetResultCheckForFailure(ActionResult<ItemStack> actionResult, EntityPlayer player, net.minecraft.world.World worldIn, ItemStack stack, EnumHand hand) {
-        // Sanity checks on the world being used (hey, i don't know the rules about clients...
-        // and if the world is in fact a responsible server world.
-        final EnumActionResult result = actionResult.getType();
-        if (!(worldIn instanceof IMixinWorld) || ((IMixinWorld) worldIn).isFake()) {
-            return result;
+    @Overwrite
+    public void onBlockClicked(BlockPos pos, EnumFacing side) {
+
+        // Sponge start - Fire interact block event
```

Comment:
If this is in fact just a pre-check, why don't we make this a cancellable injection?

Reply:
That's not possible. 
This method is also modified by Forge to add its event: https://github.com/MinecraftForge/MinecraftForge/blob/a5f29570a39508afe96870e7e665ff4bbf964aa8/patches/minecraft/net/minecraft/server/management/PlayerInteractionManager.java.patch#L34

Reply:
Then you need to communicate that in the javadoc or comments, if I did a blind sweep of overwrites, this would’ve broken because of that important fact not being documented ;)